### PR TITLE
Optimise ticklistserver

### PR DESCRIPTION
**Pull request by Spottedleaf**
See patch notes

### CODE REVIEW

**Comment by Spottedleaf** [comment]
git diff over the patch shows nothing was dropped/changed in a significant manner

File: Spigot-Server-Patches/0004-MC-Utils.patch [hunk 1]
```diff
@@ -2387,15 +2387,13 @@ index 000000000..9fa88d79d
...
 +    // assumes the sets have the same comparator, and if this comparator is null then assume T is Comparable
-+    public static <T> void mergeSortedSets(final java.util.function.Consumer<T> consumer, final java.util.SortedSet<T>...sets) {
-+        java.util.Comparator<? super T> comparator = sets[0].comparator();
-+        if (comparator == null) {
-+            comparator = (java.util.Comparator)java.util.Comparator.naturalOrder();
-+        }
++    public static <T> void mergeSortedSets(final java.util.function.Consumer<T> consumer, final java.util.Comparator<? super T> comparator, final java.util.SortedSet<T>...sets) {
 +        final it.unimi.dsi.fastutil.objects.ObjectRBTreeSet<T> all = new it.unimi.dsi.fastutil.objects.ObjectRBTreeSet<>(comparator);
-+        // note: this is done in log(n!) ~ nlogn time. It could be improved if it were to mimic what mergesort does.
++        // note: this is done in log(n!) ~ nlogn time. It could be improved if it were to mimick what mergesort does.
```
Comment (metalshark) [inline Spigot-Server-Patches/0004-MC-Utils.patch]: mimic doesn't have a k at the end

Comment (Spottedleaf) [reply Spigot-Server-Patches/0004-MC-Utils.patch]: I guess

**Review by electronicboy — APPROVED** [review]

**Comment by Spottedleaf** [comment]
on hold until a memory leak fix is ported

**Comment by Spottedleaf** [comment]
memory leak fixed ported

**Review by electronicboy — APPROVED** [review]

**Pull request merged**